client/lib/deploy.rs

```diff
@@ -510,6 +587,16 @@ mod tests {
         assert!(matches!(Deploy::read_deploy(bytes), Ok(_)));
     }
 
+    #[test]
+    fn should_round_trip_check_summed_sample_deploy_json() {
+        let bytes = CHECK_SUMMED_SAMPLE_DEPLOY.as_bytes();
+        let reader = BufReader::new(bytes);
+        let deploy: Deploy = serde_json::from_reader(reader).unwrap();
```

Comment:
Why not just `from_str`?

```suggestion
        let deploy: Deploy = serde_json::from_str(CHECK_SUMMED_SAMPLE_DEPLOY).unwrap();
```